lib/puppet/module/task.rb

```diff
@@ -157,7 +207,22 @@ def implementations
     end
 
     def files
-      implementations.map {|imp| { 'name' => imp['name'], 'path' => imp['path'] } }
+      md = metadata
+      outer_files = []
+      impl_lib_files = []
+      lib_files = []
+
+      unless md.nil?
+        outer_files = md['files'] if md.key?('files')
+        # There's definitely a more elegant way to do this...
+        if md.key?('implementations')
+          md['implementations'].each { |impl| impl_lib_files << impl['files'] if impl.key?('files') }
+        end
+        lib_files = self.class.find_files((impl_lib_files.flatten.uniq + outer_files).uniq, @module)
+      end
+      task_file = implementations.map {|imp| { 'name' => imp['name'], 'path' => imp['path'] } }
+      # PXP agent relies on 'impls' (which is the task file) being first
```

Comment:
PXP agent uses the first entry in files when there is no 'implementation' in metadata.